### PR TITLE
Add the File uploads in (a) Snap tutorial

### PR DESCRIPTION
**Pull request by jpvillaisaza**
I moved this from stackbuilders/stackbuilders.com#506. It's using an old LTS, but I think we can set a newer one.

### CODE REVIEW

**Comment by jpvillaisaza** [comment]
@jsl @seuros This tutorial is now using the latest LTS (6.5) and a slightly different approach (it was generated using literate Haskell). I think we can go ahead and have someone review it.

File: tutorials/haskell/snap-file-uploads/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+    data-files: index.html
+
+And create an executable component:
+
+    executable turbulent-sniffle
+      main-is: Main.lhs
+      other-modules: Paths_turbulent_sniffle
+      ghc-options: -Wall
+      build-depends:
```
Comment (seuros) [inline tutorials/haskell/snap-file-uploads/tutorial.md:65]: i think we need just we need just snap-server, it will require the rest.

Comment (jpvillaisaza) [inline tutorials/haskell/snap-file-uploads/tutorial.md:65]: @seuros The imports below specify what we're using from each library. snap-server installs the rest, but we have to include them here so they're not hidden.

**Comment by jpvillaisaza** [comment]
There's not a lot to change for this to work with Snap 1.0. See https://github.com/stackbuilders/turbulent-sniffle/pull/2.

**Comment by jpvillaisaza** [comment]
@jsl @mrkkrp @seuros This is now updated to work with Snap 1.0, which was released a few days ago. I think it's ready for another review and then for scheduling if it's good. Some links are not working because there's no documentation yet. I can check the links again before publishing.

File: tutorials/haskell/snap-file-uploads/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+`Main.lhs`.
+
+> To myself I sadly said,  
+> ‘She's neuralgia in her head!  
+> That dear head has nothing on it!  
+> Ought she not to wear a bonnet?’
+
+Next, let's take a look at the Haskell code in `Main.lhs`. In this case,
+we only need to export the `main` function, which is probably not the
+case for a real application:
```
Comment (mrkkrp) [inline tutorials/haskell/snap-file-uploads/tutorial.md]: In any real application `main` module exports just `main`, I'm not sure this paragraph is necessary.

File: tutorials/haskell/snap-file-uploads/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+  -> IO (Either Text ByteString)
+handleFileRead _ eitherFile =
+  case eitherFile of
+    Left policyViolation ->
+      return (Left (SnapFileUploads.policyViolationExceptionReason policyViolation))
+    Right file ->
+      fmap Right (ByteStringChar8.readFile file)
+```
+
+> Witchy kitchy kitchy wee,
```
Comment (mrkkrp) [inline tutorials/haskell/snap-file-uploads/tutorial.md]: I enjoy these too, although I'm also having hard time trying to understand if these pieces of poetry make the writing more interesting or just stay disconnected from the rest of the text producing rather strange effect.

Comment (jpvillaisaza) [inline tutorials/haskell/snap-file-uploads/tutorial.md]: @mrkkrp True. I didn't find anything good for this tutorial and the three unrelated snippets are indeed distracting. I removed them and also updated the paragraph about exporting the `main` function. Thanks!

**Pull request closed without merging**